src/methods/SANA.cpp

```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
     return highestIndex;
 }
 
-#define LOG10_LOW_TEMP -10.0
-#define LOG10_HIGH_TEMP 10.0
-#define LOG10_NUM_STEPS 20 // looping through 0 <= i <= LOG10_NUM_STEPS
+#define PBAD_HIGH_TEMP_LIMIT 0.99999
+#define PBAD_LOW_TEMP_LIMIT 1e-10
+
+
+double SANA::temperatureBracket(double LIMIT){
+	int i = 0;
+	while (true){
+		double pBadi = pForTInitial(pow(10,i));
+		double pBadi1 = pForTInitial(pow(10,i+1));
+		cout << "Testing 10 ^ " << i << " pBad = " << pBadi << endl;
+		if (pBadi < LIMIT && pBadi1 >= LIMIT){
+			break;
+		}else if (pBadi > LIMIT){
+			i--;
+		}
+		else if(pBadi < LIMIT){
+			i++;
+		}
+	}
+	return LIMIT == PBAD_HIGH_TEMP_LIMIT ? i+1 : i;
+} 
+
+double LOG10_LOW_TEMP = 0, LOG10_HIGH_TEMP = 0, LOG10_NUM_STEPS = 0;
 
 void SANA::searchTemperaturesByLinearRegression() {
 
     //if(score == "pareto") //Running in pareto mode makes this function really slow
     //	return;             //and I don't know why, but sometimes I disable using this.
     //                      //otherwise my computer is very slow.
+	
+	cout << "Testing HIGH TEMP" << endl;
+	LOG10_HIGH_TEMP = temperatureBracket(PBAD_HIGH_TEMP_LIMIT);
```

Comment:
Add second parameter here to this call, "true" for the high temp, and "false" for the lower one.